lib/share_progress/button_client.rb

```diff
@@ -0,0 +1,40 @@
+require 'share_progress'
+
+module ShareProgress
+  class ButtonClient
+    def initialize
+      @connection = ShareProgress::ConnectionFactory.get_connection
+    end
+
+    def new
+    end
+
+    def update
+    end
+
+    def find(id)
+      @connection.make_call endpoint('read'), 'get', { id: id }
+    end
+
+    def destroy(id)
+      @connection.make_call endpoint('delete'), 'post', { id: id }
+    end
+
+    def all(options={})
+      acceptable = [:offset, :limit]
+      @connection.make_call endpoint, 'get', filter_keys(options, acceptable)
+    end
```

Comment:
I've seen enough direct calls to HTTParty in similar repos I think I feel comfortable doing it like this, especially since we'd still be able to preserve the same `Client.get '/buttons'...` API by changing the call in client if need be.